src/arrayMethodSort.js

```diff
@@ -5,7 +5,27 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    const compare = compareFunction || function(a, b) {
+      if (a < b) {
+        return -1;
+      }
+
+      if (a > b) {
+        return 1;
+      }
+
+      return 0;
+    };
+
+    for (let i = 0; i < this.length - 1; i++) {
+      for (let q = 0; q < this.length - 1 - i; q++) {
+        if (compare(this[q].toString(), this[q + 1].toString()) > 0) {
```

Comment:
In the compare function, the elements are always converted to strings. It should only convert to string when no compareFunction is provided.

Reply:
Yes, you need to use String in your compareFunction()